test/batch/test_crud_batch.py

```diff
@@ -81,6 +88,56 @@ def test_timeout_retries(self):
         self.check_instance(batch, timeout_retries=1)
         check_error_message(self, error, type_error)
 
+    def test_connection_error_retries(self):
+        """
+        Test Setter and Getter for 'connection_error_retries'.
+        """
+
+        batch = Batch(Mock())
+        self.check_instance(batch=batch)
+
+        self.assertEqual(batch.connection_error_retries, 3)
+        self.check_instance(batch, connection_error_retries=3)
+
+        batch.connection_error_retries = 10
+        self.assertEqual(batch.connection_error_retries, 10)
+        self.check_instance(batch, connection_error_retries=10)
+
+        batch.connection_error_retries = 0
+        self.assertEqual(batch.connection_error_retries, 0)
+        self.check_instance(batch, connection_error_retries=0)
+
+        batch.connection_error_retries = 1
+        self.assertEqual(batch.connection_error_retries, 1)
+        self.check_instance(batch, connection_error_retries=1)
+
+        # exceptions
+        ## error messages
+        value_error = "'connection_error_retries' must be positive, i.e. greater or equal that zero (>=0)."
+        type_error = f"'connection_error_retries' must be of type {int}."
+
+        #######################################################################
+        # test wrong value
+        with self.assertRaises(ValueError) as error:
+            batch.connection_error_retries = -1
+        self.assertEqual(batch.connection_error_retries, 1)
+        self.check_instance(batch, connection_error_retries=1)
+        check_error_message(self, error, value_error)
+
+        #######################################################################
+        # test wrong type
+        with self.assertRaises(TypeError) as error:
+            batch.connection_error_retries = True
+        self.assertEqual(batch.connection_error_retries, 1)
+        self.check_instance(batch, connection_error_retries=1)
+        check_error_message(self, error, type_error)
+
+        with self.assertRaises(TypeError) as error:
+            batch.connection_error_retries = '2'
+        self.assertEqual(batch.connection_error_retries, 1)
+        self.check_instance(batch, connection_error_retries=1)
+        check_error_message(self, error, type_error)
```

Comment:
You could make this much shorther with pytest + parametrization (unittest also has something)

```
    @pytest.mark.parametrization("error_retries", [10, 0, ... ]
    def test_connection_error_retries(error_retries):
        """Test Setter and Getter for 'connection_error_retries'."""

        batch = Batch(Mock())
        self.check_instance(batch=batch)

        assert batch.connection_error_retries == 3
        self.check_instance(batch, connection_error_retries=3) <= that would need to be moved out of the unittest class, but it can still be moved from everywhere

        batch.connection_error_retries = error_retries
        self.assertEqual(batch.connection_error_retries, error_retries)
        self.check_instance(batch, connection_error_retries=error_retries)
```

and for errors:
    @pytest.mark.parametrization("error_retries,error", [(-1, Valueerror), 0, ... ]
    def test_connection_error_retries_exceptions(error_retries, error):
        batch = Batch(Mock())
        check_instance(batch=batch)
        with pytest.raises(error) as excinfo:
            .....

Reply:
Lets leave it as it is for now because the code is different a bit for different timeout retries (one with `objects` another with `references`).

Reply:
oki